src/Symfony/Contracts/Deprecation/README.md

```diff
@@ -0,0 +1,22 @@
+Symfony Deprecation Contracts
+=============================
+
+A generic function and convention to trigger deprecation notices.
+
+This package provides a single global function named `deprecated()`.
+Its purpose is to trigger deprecations in a way that can be silenced on production environment
+by using the `zend.assertions` ini setting and that can be caught during development to generate reports.
+
```

Comment:
```suggestion

**NOTE:** By disabling `zend.assertions` all `assert()` method calls in your code will not work anymore!
```

@nicolas-grekas sth like this.

TLDR:
One think he will JUST silence deprecations but could deactivate important assert()-calls in his code! I want to make the user aware of this! Thats all

Reply:
That is way too scary.
People that use `assert()` already know that the only reason why they do so is to benefit from this feature of PHP.
The very promise and the only use case for `assert()` is to disable-able in prod to turn dev-only checks into no-op in prod.
We're now going to teach against this feature.

Reply:
Can't agree more with @nicolas-grekas: the very purpose of `assert` in PHP is that it has no side-effects whatsoever in production, not even the overhead of the method call to `trigger_error`. That makes it the ideal solution. People that use `assert` for other purposes (e.g. to force strict type checking in production) should be aware of what enabling `zend.assertions` on production environments entails. We should neither advocate that use-case, nor cater to it.